src/Configuration/ConfigurationFactory.php

```diff
@@ -255,4 +261,28 @@ private static function retrieveMinCoveredMsi(?float $minCoveredMsi, SchemaConfi
     {
         return $minCoveredMsi ?? $schema->getMinCoveredMsi();
     }
+
+    /**
+     * @param array<string, mixed[]> $resolvedMutatorsMap
+     *
+     * @return array<string, array<int, string>>
+     */
+    private function retrieveIgnoreSourceCodeMutatorsMap(array $resolvedMutatorsMap): array
+    {
+        $map = [];
+
+        foreach ($resolvedMutatorsMap as $mutatorClassName => $config) {
+            if (array_key_exists('ignoreSourceCodeByRegex', $config)) {
+                $classNameParts = explode('\\', $mutatorClassName);
```

Comment:
There's a trait `GetMutatorName`, if `getName()` can be made static, and I think it is, then this could be greatly simplified by a call to `$mutatorClassName::getName()`

Reply:
> Make `Mutator::getName()` non-static #989 

🤣 

I'm personally ok with 2 duplicated lines here

Reply:
Well, one can totally do just that: https://3v4l.org/XNklO

Reply:
#1331

Reply:
updated